src/Microsoft.AspNetCore.Dispatcher/DependencyInjection/DefaultDispatcherConfigureOptions.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Dispatcher;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    internal class DefaultDispatcherConfigureOptions : IConfigureOptions<DispatcherOptions>
```

Comment:
This thing will auto-configure `DispatcherOptions` based on DI. The separate interfaces are needed because because the delegate types we use on dispatcher options aren't DI friendly.

Basically you need to register a class, but we need to retrieve a delegate. Extra interfaces are needed.

Reply:
This would work pretty well for something like MVC's model binders and formatters where the options are really doing a job that DI could already do.